output/cloud/expv2/collect.go

```diff
@@ -0,0 +1,158 @@
+package expv2
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"go.k6.io/k6/metrics"
+	"go.k6.io/k6/output"
+)
+
+type bucketQ struct {
+	m       sync.Mutex
+	buckets []timeBucket
+}
+
+// Buckets pop all the queued buckets.
+func (q *bucketQ) Buckets() []timeBucket {
+	q.m.Lock()
+	defer q.m.Unlock()
+
+	if len(q.buckets) < 1 {
+		return nil
+	}
+	b := make([]timeBucket, len(q.buckets))
+	copy(b, q.buckets)
+	q.buckets = q.buckets[:0]
+	return b
+}
+
+// Insert adds an item in the queue.
+func (q *bucketQ) Push(b []timeBucket) {
+	q.m.Lock()
+	q.buckets = append(q.buckets, b...)
+	q.m.Unlock()
+}
+
+type timeBucket struct {
+	TimeSeries metrics.TimeSeries
+	Time       time.Time
```

Comment:
I don't think we need those as this is information that already exists and this way it will be copied a bunch of times.

Reply:
How do we pass them to the flush operation? I can drop them during the _aggregation_ phase because they are already in the _perSeriesBuckets_ so we can build this struct only before enqueueing. But we can't skip doing it before calling `bucketQ.Push`.
Am I missing something?

Reply:
Ah, I'd missed @codebien's comment when writing my own, so it's not a response to it but to @mstoykov's.